test/tests_for_sdkmr/ee2_SDKMethodRunner_test_EE2Runjob_test.py

```diff
@@ -247,6 +247,189 @@ def test_run_job(self, rq_mock, condor_mock):
+                retry_from_retry_id,
+                retry_from_original_again,
+            ]
+        )["job_states"]
+
+        for job in [original_job, retried_job, retried_job2, retried_job3]:
+            if job == original_job:
+                assert original_job["retry_count"] == 3
+            else:
+                assert job["retry_parent"] == parent_job_id
+
+        # 4. Get jobs and ensure they contain the same keys and params
+        same_keys = ["user", "authstrat", "wsid", "scheduler_type", "job_input"]
+
+        assert "retry_parent" not in original_job
+
+        for key in same_keys:
+            assert original_job[key] == retried_job[key]
+
+        assert original_job["job_input"]["params"] == retried_job["job_input"]["params"]
+
+        # TODO Retry a job that uses run_job_batch or kbparallels (Like metabat)
+        # TODO Retry a job without an app_id
+
+    @requests_mock.Mocker()
+    @patch("lib.execution_engine2.utils.Condor.Condor", autospec=True)
+    def test_retry_job_with_params_and_nci_and_src_ws_objs(self, rq_mock, condor_mock):
+        # 1. Run the job
+        rq_mock.add_matcher(
+            run_job_adapter(
+                ws_perms_info={"user_id": self.user_id, "ws_perms": {self.ws_id: "a"}}
+            )
+        )
+        runner = self.getRunner()
+        runner.workspace.get_object_info3 = MagicMock(return_value={"paths": []})
+        runner.workspace_auth.can_write = MagicMock(return_value=True)
+        runner.get_condor = MagicMock(return_value=condor_mock)
+
+        quast_params = {
+            "workspace_name": "XX:narrative_1620418248793",
+            "assemblies": ["62160/9/18"],
+            "force_glimmer": 0,
+        }
+        source_ws_objects = quast_params["assemblies"]
+        nci = {
+            "run_id": "3a211c4e-5ba8-4b94-aeae-378079ccc63d",
+            "token_id": "f38f09f7-5ab1-4bfc-9f3f-2b82c7a8dbdc",
+            "tag": "release",
+            "cell_id": "3ee13d64-623b-407f-98a1-72e577662132",
+        }
+
+        job = get_example_job_as_dict(
+            user=self.user_id,
+            wsid=self.ws_id,
+            narrative_cell_info=nci,
+            params=quast_params,
+            source_ws_objects=source_ws_objects,
+            method_name="kb_quast.run_QUAST_app",
+            app_id="kb_quast/run_QUAST_app",
+        )
+        si = SubmissionInfo(clusterid="test", submit=job, error=None)
+        condor_mock.run_job = MagicMock(return_value=si)
+        parent_job_id = runner.run_job(params=job)
+
+        # 2. Retry the job
+        runner.update_job_status(job_id=parent_job_id, status=Status.terminated.value)
+        retry_job_id = runner.retry(job_id=parent_job_id)["retry_id"]
+
+        # 3. Get both jobs and compare them!
+        original_job, retried_job = runner.check_jobs(
+            job_ids=[parent_job_id, retry_job_id]
+        )["job_states"]
+
+        same_keys = ["user", "authstrat", "wsid", "scheduler_type", "job_input"]
```

Comment:
shouldn't this test the method, app id, params, nci, and source ws objs?

Reply:
by doing the assert on job_input, it checks to see if they are the same

Reply:
Ugh, right :+1: